text/0419-cdk-launchpads.md

```diff
@@ -0,0 +1,629 @@
+> but I'm only showing the standalone version for brevity.
+
+#### 1. `status`
+
+Shows the current status of any Launchpads defined in the given environment.
+
+This invocation will use the current AWS credentials in the console:
+
+```shell
+$ cdk-lpads status
+
+Environment - account: 123, region: us-east-1
+1 Launchpad found:
+1 | default CDK bootstrap Stack | version: 10   (UP-TO-DATE)
+```
+
+You can also check a specific account and region, using a profile defined in your AWS config files:
+
+```shell
+$ cdk-lpads status —profile acc-456 aws://456/eu-west-2
+
+Environment - account: 456, region: eu-west-2
+1 Launchpad found:
+1 | legacy (V1) CDK bootstrap Stack
+    Run `cdk-lpads update` to upgrade it to the newest version.
+```
+
+#### 2. `new`
+
+Creates a new Launchpad in the given account and region.
+By default, the command will include a wizard that will guide the user through the options:
+
+```shell
+$ cdk-lpads new
+
+Welcome! This wizard will guide you through creating a new Launchpad in the environment.
+To start, let's make sure we're operating in the correct AWS account and region:
+
+Environment - account: 123, region: eu-west-2
+
+Does this look correct? [Y/N] Y
+
+Looks like the environment aws://123/eu-west-2 doesn't have any existing Launchpads.
+
+Do you want to customize the qualifier used? [default: hnb659fds]:
+...
+```
+
+Alternatively, the `new` command can be customized directly, instead of going through the wizard;
+see "Customizing the Launchpad" below.
+
+#### 3. `update`
+
+This command attempts to update an existing Launchpad.
+In the case it's the default CDK bootstrap Launchpad, it will simply update it to the latest version.
+The `update` command can be customized directly, same as the `new` command;
+see "Customizing the Launchpad" below.
+
+#### 4. `diff`
+
+Allows you to see what differences would be applied if you updated your Launchpad.
+Useful to run before `update` to confirm the changes are what you expect.
+
+#### 5. `init json/yaml/cdk [ts/js/java/python/csharp/go]`
+
+Similar to the `cdk init` command,
+this command creates a skeleton that allows you get started with customizing your Launchpad.
+It supports both the JSON/YAML and CDK ways of customizing Launchpads (see below).
+
+### Ways of customizing the Launchpad
+
+The default Launchpad contains the same resources as the `cdk bootstrap` command (in version 2) creates.
+A CDK Launchpad can be customized with the following additional options:
+
+1. Compliance rules.
+2. Additional resources (deployed using CloudFormation, CDK, or in any other way, and simply referenced by the Launchpad).
+3. Modifications to the CDK bootstrap resources (using either CloudFormation templates or CDK).
+4. Setting IAM Permission Boundaries used by all applications deployed into a given environment.
+
+There are two ways of customizing a Launchpad: using JSON/YAML files, or using CDK code.
+
+#### Customizing the Launchpad with JSON/YAML files
```

Comment:
Yes, a section of the JSON/YAML file will be translated to a CloudFormation template. We need our own JSON/YAML structure for all of those things that are not part of CloudFormation, like Compliance rules, and shared exported resources (I mentioned them in https://github.com/aws/aws-cdk-rfcs/pull/421#discussion_r861169550).

Reply:
To echo this concern, I'd say I'd still maybe separating those out as otherwise you end up with a bizzare `JSON/YAML` that ends up having 2-3 sub-DSLs because of all the knob it turns. On it's face it makes it simple (on file) but I think you're aggregating complexity.

Reply:
That's true. Maybe having multiple separate files, and gluing them with CLI options, is the better design here.

Reply:
I feel like that would be a ton of work on the CDK team's side to own that resource, so I'd rather avoid it, at least in the first iteration (we can transition to that model once we prove the concept out more).